AirSkyBoat/AirSkyBoat
# Garrison System

**_I affirm:_**

## What does this pull request do?

* Implements wave system that spawns the right amount of mobs at the right time for each wave
* Fixes the number of allies spawned (1 per alliance member based on captures)
* Grants correct loot on victory
* Spawns the right bosses / npcs on each zone
* Save lockout information for player
* Add timer for garrison timeout
* Gates entry based on rank and other requirements

## Known Issues

* NPC looks are incorrect (they show naked on spawn). This seems like an entity update issue.
* Some zones (Qufim, buburimbu) have an issue where npcs will path incorrectly to mobs and never attack. This is a bigger issue that is not related to the garrison system so I'm not fixing it here, but it makes certain zones harder to play.
* On victory / loss, the win / lose message is shouted out by the npc immediately. This is not accurate to captures, as the npc is supposed to tell a player that message after the player talks to the npc on victory / loss. However, this felt like a fair simplification given the little impact on players.

## Steps to test these changes

* edit settings/main.lua and set ENABLE_GARRISON to true. Maybe set minimum rank to 0 and remove nation lockout for testing. should also set DEBUG_GARRISON=true
* !zone 103
* Run to op, !additem sack_of_galka% and trade it to the npc maybe use !immortal on the spawned ally if you suck like I do
* Kill mobs as they pop
